lib/Dialect/ESI/runtime/cpp/tools/esiquery.cpp

```diff
@@ -0,0 +1,46 @@
+//===- esiquery.cpp - ESI accelerator system query tool -------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// DO NOT EDIT!
+// This file is distributed as part of an ESI package. The source for this file
+// should always be modified within CIRCT
+// (lib/dialect/ESI/runtime/cpp/lib/backends/Cosim.cpp).
+//
+//===----------------------------------------------------------------------===//
+
+#include "esi/Accelerator.h"
+
+#include <iostream>
+#include <map>
+#include <stdexcept>
+
+using namespace esi;
+
+int main(int argc, const char *argv[]) {
+  // TODO: find a command line parser library rather than doing this by hand.
```

Comment:
Be lazy and just use JSON

Reply:
See my previous comment about JSON library. I also don't think JSON makes sense for command line interactions. I wouldn't want to have to write JSON to just to query the perf counters.